app/src/main/java/com/example/notes2text/adapters/fragments/DirectoryAccessController.java

```diff
@@ -163,13 +164,25 @@ public boolean onMenuItemSelected(@NonNull MenuItem menuItem) {
                     }
                     directoryPresenter.BackLayerSuccess(getActivity());
                 } else if (menuItem.getItemId() == R.id.create_folder_button) {
-                    directoryPresenter.FolderCreationSuccess(getActivity());
+                    //create a new dialogue using openDialog when button is clicked.
+                    openDialog();
                 }
                 return true;
             }
         }, getViewLifecycleOwner(), Lifecycle.State.RESUMED);
     }
 
+    /**
+     * Helper method that create a new CreateFolderController, which inherits
+     * AppCompatDialogFragment and can create a new dialogue.
+     */
+    public void openDialog() {
```

Comment:
Should this be public? Make it private if there isn't need to access it from outside.

Reply:
Changed the method to private in the new commit